### PR TITLE
Namco System 21 and other related refactors

### PR DESCRIPTION
**Pull request by DavidHaywood**
note this includes cam900s changes and thus supersedes his submission ( https://github.com/mamedev/mame/pull/3999 ) as it was working on the same code.

 - gave an option for tms320c25 to use an internal ROM config (it's controlled by a pin)
 - converted Namco C67 to be a tms320c25 type with internal ROM
 - reorganized and split out the older and newer System 21 DSP boards into their own board devices (this can now be shared with Galaxian 3 which has 2 DSP boards identical to the later System 21 one)
 - split out the 3D renderer into a device too (likewise, this will enable us to use it with Galaxian 3)
 - corrected bad documentation (I've contacted owners of 3 Star Blade PCBs and in all cases their boards have been identical to Cyber Sled, the configuration previously described does not appear to exist and is not logical)
 - various cleanups (eg suspend the DSP CPU in the earlier S21 type rather than running a dummy program before the 68k uploads to RAM)

I consider this to be a checkpoint that will serve as the base for futher cleanups (and work on things like Galaxian 3 + the proper Drivers Eyes PCB cage setup, which I'll look into soon)

### CODE REVIEW

File: src/devices/cpu/tms32025/tms32025.cpp [hunk 1]
```diff
@@ -184,7 +184,16 @@ Table 3-2.  TMS32025/26 Memory Blocks
 #define IND     m_AR[ARP]                       /* address used in indirect memory access operations */
 
 
+/*
+   Processor can be operated in one of two modes based on Pin 1 (MP/MC)
+   MP/MC = 1 (Microprocessor Mode)
+   MP/MC = 0 (Microcomputer Mode)
+   in 'Microcomputer' mode the 4K Word internal ROM is used (TMS320C25)
+
+   using a different device type for each as MAME doesn't appear to offer an easy way to handle this otherwise?
```
Comment (happppp) [inline src/devices/cpu/tms32025/tms32025.cpp]: That's not a very nice workaround. What methods did you try? If there really is no other way, can't you pretend this internal rom is external?

Comment (DavidHaywood) [reply src/devices/cpu/tms32025/tms32025.cpp]: I'd rather not pretend the internal ROM was external, IMHO that's an uglier workaround.

I could see no way to have a different internal map depending on the configuration however.  It's fixed configuration afaik (you can't change the pin state after boot) so this seemed the most logical way.

Comment (DavidHaywood) [reply src/devices/cpu/tms32025/tms32025.cpp]: the only other workaround seem to actually just be exploiting bugs in MAME with regards external ROMs mapping over internal ROMs and I'd *really* rather not do that

Comment (DavidHaywood) [reply src/devices/cpu/tms32025/tms32025.cpp]: so yes, for the time being I think this is the cleanest solution by far

Comment (DavidHaywood) [reply src/devices/cpu/tms32025/tms32025.cpp]: unless I install ROM in the device, but that seems to be frowned upon these days?  would resolve your issue tho.

Comment (DavidHaywood) [reply src/devices/cpu/tms32025/tms32025.cpp]: well I've done that, not sure I'm actually happier with that solution, but it still works without breaking everything elsewhere or relying on weird side-effects.

**Comment by Hydreigon223** [comment]
I hope some machines will be promoted to working after these changes and more. Would also like to see network emulation sometime.

**Comment by DavidHaywood** [comment]
well modernizing things and untangling them should make the drivers easier to work with, as their existing state was rather offputting.  it also makes it more obvious where things are hacked, rather than being emulated properly.  these things might result in it being easier for others to improve the emulation.

**Comment by Hydreigon223** [comment]
I guess this explains some weird stuff with Namco System 21 prior to these changes. Idk if Namco System 21 linking feature is the same used in System 2 and System 22. A Namco c139 serial controller? Idk what type of cable it uses either.

I myself like asking these linking related questions.

**Comment by sjy96525** [comment]
Driver is updated, but Cyber Sled (Japan) still show the English text in select screen and attract mode, and hasn't notice screen, can you help?
https://i.imgur.com/yVtdBN3.png
https://i.imgur.com/kzQK5kl.png
https://i.imgur.com/SkM0Nd7.png

**Comment by sjy96525** [comment]
I play the Driver's Eyes has Receive Error screen for Left and Right screen.

https://i.imgur.com/xhg48IJ.png

**Comment by DavidHaywood** [comment]
drivers eye's isn't marked as working, the issue you report is already known and listed.  if you're going to start spamming again then this work stops as quickly as it started.

**Comment by DavidHaywood** [comment]
anyway, could do with this merging at this point.

there's potential cleanup / de-duplication work to be done across namcos2 / namcos21, but the bulk of it is here now.

**Pull request merged**